library/Sexp/src/Juvix/Sexp.hs

```diff
@@ -98,6 +103,40 @@ foldSearchPred t p1@(predChange, f) p2@(predBind, g) =
       | predChange ":atom" -> f a t
       | otherwise -> pure $ Atom a
 
+-- | @foldSearchPredManualRecurse@ is just like @foldsSearchPred@
+-- except the function which changes the sexp form manually has to
+-- recurse, it no longer automatically does this.
```

Comment:
There does seem to be significant overlap in the code -- is it possible to factor out the commonality between them?

Reply:
Probably can make one in the other, by making a new f such that it always recurses

Reply:
That'd probably be a good thing, then, although I wouldn't say it needs to be done for this PR, maybe just for some low-priority "refactoring"-tag issue afterwards.